test/contract_tests.js

```diff
@@ -0,0 +1,543 @@
+            await Einstance.connect(Client).deposit(reportRoot, NativeAsset, 100, { value: 100 })
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, NativeAsset, 0, salt, reportAddress, merkleProofval));
+        })
+
+        it("withdraw(): revert if withdrawing with invalid Payment/Token Address",async function(){
+            await Einstance.connect(Client).deposit(reportRoot, NativeAsset, 100, { value: 100 })
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee, 10, salt, reportAddress, merkleProofval)).to.be.reverted;
+        })
+
+        it("withdraw(): revert if withdrawal amount > report amount", async function(){
+            await Einstance.connect(Client).deposit(reportRoot, NativeAsset, 100, { value: 100 })
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, NativeAsset, 1000, salt, reportAddress, merkleProofval)).to.be.reverted;
+        })
+
+        it("withdraw(): Anyone can perform withdraw() on the report", async function () {  // but the payments goes to original report owner who provided walletaddress
+            await Einstance.connect(Client).deposit(reportRoot, NativeAsset, ONE_ETHER_FORMAT, { value: ONE_ETHER_FORMAT })
+            await expect(Einstance.connect(Triager).withdraw(reportRoot, NativeAsset, HALF_ETHER_FORMAT, salt, reportAddress, merkleProofval))
+        })
+
+        it("[TODO] withdraw(): No underflow/overflow on amount", async function () {
+            // TODO
+        })
+    })
+
+    describe("===> updateReport()",function(){
+        let getReportRoot,
+            key,
+            commitment,
+            attest_id,
+            newStatusBitField;
+
+        beforeEach(async () => {
+            await instance.connect(Triager).submit(F_SUBMIT(report1));
+            [getReportRoot, key, commitment] = F_attest(report1, Triager.address);
+
+            newStatusBitField = 00000001; // Bitnumbers
+            Rkey = ReportKeys.report;
+            attest_id = F_getAttestionID(report1, Triager.address, Rkey);
+        })
+
+        it("updateReport(): Update the report with newStatusBitField",async function(){
+            instance.connect(Triager).attest(getReportRoot, key, commitment)
+            await expect(instance.connect(Triager).updateReport(getReportRoot, newStatusBitField));
+        })
+
+        it("updateReport(): Update the report with newStatusBitField and check ReportStatus",async function(){
+            instance.connect(Triager).attest(getReportRoot, key, commitment)
+
+             // Bit Fields binary: 0 here is to get the `first` bit from 8 bits , i.e 1 byte = 8 bits, since boolean use only first bit
+
+            // false: report has not been updated
+            await expect(await instance.connect(Triager).reportHasStatus(getReportRoot, Triager.address, 0)).to.be.false;
```

Comment:
We should consider removing this function altogether because it's a big footgun. We should encourage people to use the `validateXXX` methods instead.